### ISSUE TITLE
mysql keyword rank

### ISSUE THREAD

**Issue by lhovo** [biopython/biopython#1882]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

```3.6.7 (default, Dec 20 2018, 01:14:33) 
[GCC 4.2.1 Compatible FreeBSD Clang 6.0.0 (tags/RELEASE_600/final 326565)]
CPython
FreeBSD-11.2-RELEASE-p4-amd64-64bit-ELF
1.73

mysql  Ver 8.0.12 for FreeBSD11.2 on amd64 (Source distribution)
```

Running code

```from Bio import Entrez
from Bio import SeqIO
handle = Entrez.efetch(db="nuccore", id="6273291,6273290,6273289", rettype="gb", retmode="text")
for seq_record in SeqIO.parse(handle, "genbank") :
    print(seq_record.id, seq_record.description[:50] + "...")
    print("Sequence length %i," % len(seq_record.seq),)
    print("from: %s" % seq_record.annotations['source'])
handle.close()
```

### Expected behaviour

To insert records as per example
https://biopython.org/wiki/BioSQL

### Actual behaviour

Failed to insert record due to table name "rank", is now a keyword in mysql?

### Steps to reproduce

I believe in mysql 8.0 'rank' is now a keyword,
by escaping all instances of `rank` in https://github.com/biopython/biopython/blob/master/BioSQL/Loader.py
I was able to get this example to work.

Unsure if this will cause issues with other sql databases if the Insert table names are escaped.

**Comment by peterjc** [comment]
This makes sense, according to https://dev.mysql.com/doc/refman/8.0/en/keywords.html

> RANK (R); added in 8.0.2 (reserved)

How exactly did you escape the table's column name? In words or better yet as a pull request would be great?

If this is simple we probably don't need to change the schema itself - which would be annoying as BioSQL is shared across Bio* projects.

**Comment by lhovo** [comment]
Sorry my code was rather hackey as I only escaped the rank key words.

I would expect to do this correctly the code should escape all table names like so.

```
sql = "INSERT INTO bioentry_reference (`bioentry_id`, `reference_id`," \
              " `start_pos`, `end_pos`, `rank`) VALUES (%s, %s, %s, %s, %s)"
```

**Comment by peterjc** [comment]
If we just wrap the problematic column name ``rank`` with back-ticks that's good news. We won't need to change the schema.

I've logged https://github.com/biosql/biosql/issues/22 on the schema itself, but it seems likely we need to make this change to lots of our Biopython SQL command with the column name rank in them - in both ``BioSQL/BioSeq.py`` and ``BioSQL/Loader.py``

Are you interested in making and testing these changes yourself, and submitting a pull request?

**Comment by lhovo** [comment]
I was helping a friend get this working for their project as such I'm not overly knowledgeable in bio concepts.

I'm about to go on holidays so it would be 2+ months before I would get around making the changes.

If the work hasn't started before then I will give it a go, it should be straight forward but finding all the SQL queries in the code and testing them might take time as I'm unfamiliar with the code and how it works / what to test for.

**Comment by peterjc** [comment]
Thanks. I'll tag this issue as help wanted for now.

Based on our test coverage measurements, although BioSQL is not fully covered, looking at https://codecov.io/gh/biopython/biopython/tree/master/BioSQL by eye suggests all most all the SQL lines are checked. So, the good news is getting our test suite to pass should be enough - once you've managed to get MySQL v8 installed and configured ``Tests/biosql.ini`` appropriately.

**Comment by peterjc** [comment]
As noted by Hilmar Lapp on https://github.com/biosql/biosql/issues/22 using double quotes for escaping would be more standard (it seems the backticks may be MySQL specific).

**Comment by nimne** [comment]
I'll take a look at this issue! MySQL 8 is considered a stable production release so it should certainly be supported.

**Comment by peterjc** [comment]
This should be fixed by pull request #1934, closing.
